src/interface_vault_operator_peers.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Work in progress interface for Vault peer relations."""
+
+import logging
+
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Relation
+
+
+class HasPeersEvent(EventBase):
+    """Has peers event."""
+
+    pass
+
+
+class ReadyPeersEvent(EventBase):
+    """Ready Peers event."""
+
+    pass
+
+
+class VaultOperatorPeerEvents(ObjectEvents):
+    """Vault Operator Peer events."""
+
+    has_peers = EventSource(HasPeersEvent)
+    ready_peers = EventSource(ReadyPeersEvent)
+
+
+class VaultOperatorPeers(Object):
+    """Handles logic between instances of the vault operator."""
+
+    ROOT_TOKEN = "root-token"
+    ROOT_CA = "root-ca"
+    UNSEAL_KEY = "unseal-key"
+
+    on = VaultOperatorPeerEvents()
+
+    def __init__(self, charm, relation_name):
```

Comment:
This and a few others in this file could warrant some more typing and better docstrings documenting return types/params

Reply:
Agreed, fixed